File: Day 15 : Linked List/Solution.rb
class Node
    attr_accessor :data,:next

    def initialize data
        @data = data
        @next = nil
    end
end

class Solution
  def insert(head,value)
    node = Node.new(value)
    if head == nil
      head = node
    else
      curr = head
      while curr.next
          curr = curr.next 
      end
      curr.next = node
    end
    return head
  end
  
  def display(head)
    current = head
    while current 
      print current.data," "
      current = current.next
    end
  end
end
       
mylist = Solution.new
head = nil
T = gets.to_i
for i in 1..T
    data = gets.to_i
    head = mylist.insert(head,data)
end
mylist.display(head)

File: Day 15 : Linked List/Solution.ts
class MyNode {
    data: number
    next: MyNode | null

    constructor(data: number, next: MyNode | null) {
        this.data = data;
        this.next = next;
    }
}

class Solution {
    insert(head: MyNode | null, data: number): MyNode | null {
        let node = new MyNode(data, null);
        node.data = data;

        if (head == null) {
            head = node;
        } else {
            let current = head;
            while (current.next) {
                current = current.next;
            }
            current.next = node;
        }

        return head;
    }

    display(head: MyNode | null): void {
        let start = head;
        while (start) {
            process.stdout.write(start.data + " ");
            start = start.next;
        }
    }
}

function main() {
    const T: number = +readLine();
    let head = null;
    let mylist = new Solution();
    for (let i: number = 0; i < T; i++) {
        let data: number = +readLine();
        head = mylist.insert(head, data);
    }
    mylist.display(head);
}